cpp/ql/test/query-tests/Security/CWE/CWE-190/semmle/TaintedAllocationSize/test.cpp

```diff
@@ -139,7 +139,7 @@ void more_bounded_tests() {
 
 		if (size > 0)
 		{
-			malloc(size * sizeof(int)); // BAD
+			malloc(size * sizeof(int)); // GOOD (overflow not possible)
```

Comment:
Could we add a test with a `long` to keep testing `size > 0` with an actual overflow?

Reply:
Good point. I've done this in 31091c66c19fdb401967decc596621fa68db8d0a. Sorry about the horrible diff from that commit. (I've checked that it does the right thing.)